src/dbnode/storage/bootstrap/process.go

```diff
@@ -372,3 +376,66 @@ func (b bootstrapProcess) newRunOptions() RunOptions {
 		).
 		SetInitialTopologyState(b.initialTopologyState)
 }
+
+// NewNamespaces returns a new set of bootstrappable namespaces.
+func NewNamespaces(
+	namespaces []ProcessNamespace,
+) Namespaces {
+	namespacesMap := NewNamespacesMap(NamespacesMapOptions{})
+	for _, ns := range namespaces {
+		namespacesMap.Set(ns.Metadata.ID(), Namespace{
+			Metadata:        ns.Metadata,
+			Shards:          ns.Shards,
+			DataAccumulator: ns.DataAccumulator,
+		})
+	}
+	return Namespaces{
+		Namespaces: namespacesMap,
+	}
+}
+
+// NewNamespaceResults creates a
+// namespace results map with an entry for each
+// namespace spoecified by a namespaces map.
+func NewNamespaceResults(
+	namespaces Namespaces,
+) NamespaceResults {
+	resultsMap := NewNamespaceResultsMap(NamespaceResultsMapOptions{})
```

Comment:
Can probably init this with size too
```
	nsMap := namespaces.Namespaces.Iter()
	resultsMap := NewNamespaceResultsMap(NamespaceResultsMapOptions{
		InitialSize: len(nsMap),
	})
	for _, entry := range nsMap {
```

Reply:
Can't be bothered when it's in such a low frequency call code path TBH.